static/app/components/core/button/index.tsx

```diff
@@ -110,6 +112,165 @@ interface ButtonPropsWithAriaLabel extends BaseButtonProps {
+              props.priority && ['primary', 'danger'].includes(props.priority)
+            }
+          />
+        )}
+        <ButtonLabel size={size} borderless={props.borderless}>
+          {props.icon && (
+            <Icon size={size} hasChildren={hasChildren}>
+              <IconDefaultsProvider size={ICON_SIZES[size]}>
+                {props.icon}
+              </IconDefaultsProvider>
+            </Icon>
+          )}
+          {props.children}
+        </ButtonLabel>
+      </StyledButton>
+    </Tooltip>
+  );
+}
+
+export const StyledButton = styled('button')<ButtonProps>`
+  ${p => (p.theme.isChonk ? getChonkButtonStyles(p as any) : getButtonStyles(p as any))}
+`;
+
+export interface LinkButtonProps
+  extends CommonButtonProps,
+    Omit<LinkElementProps, 'role'> {
+  /**
+   * Determines if the link is disabled.
+   */
+  disabled?: boolean;
+  /**
+   * Determines if the link is external and should open in a new tab.
+   */
+  external?: boolean;
+  href?: string;
+  /**
+   * If true, the link will not reset the scroll position of the page when clicked.
+   */
+  preventScrollReset?: boolean;
+  /**
+   * Determines if the link should replace the current history entry.
+   */
+  replace?: boolean;
+  to?: string | LocationDescriptor;
+}
+
+export function LinkButton({
+  size = 'md',
+  to,
+  href,
+  disabled,
+  tooltipProps,
+  ...props
+}: LinkButtonProps) {
+  const {handleClick, hasChildren, accessibleLabel} = useButtonFunctionality({
+    ...props,
+    to,
+    href,
+    disabled,
+  });
+
+  return (
+    <Tooltip skipWrapper {...tooltipProps} title={props.title} disabled={!props.title}>
+      <StyledLinkButton
+        aria-label={accessibleLabel}
+        aria-disabled={disabled}
+        size={size}
+        href={disabled ? undefined : href}
+        to={disabled ? undefined : to}
+        disabled={disabled}
+        {...props}
+        onClick={handleClick}
+      >
+        {props.priority !== 'link' && (
+          <InteractionStateLayer
+            higherOpacity={
+              props.priority && ['primary', 'danger'].includes(props.priority)
+            }
+          />
+        )}
+        <ButtonLabel size={size} borderless={props.borderless}>
+          {props.icon && (
+            <Icon size={size} hasChildren={hasChildren}>
+              <IconDefaultsProvider size={ICON_SIZES[size]}>
+                {props.icon}
+              </IconDefaultsProvider>
+            </Icon>
+          )}
+          {props.children}
+        </ButtonLabel>
+      </StyledLinkButton>
+    </Tooltip>
+  );
+}
+
+const StyledLinkButton = styled(
+  ({size: _size, title: _title, external, ...props}: LinkButtonProps) => {
+    if (props.to) {
+      return <Link {...props} to={props.to} role="button" />;
+    }
+
+    if (props.href) {
+      return (
+        <a
+          {...props}
+          {...(external ? {target: '_blank', rel: 'noreferrer noopener'} : {})}
+          role="button"
+        />
+      );
+    }
```

Comment:
could we try and separate this inline component from `StyledLinkButton`, as this is logic that isn’t concerned with styling. I think it’s the main reason why we need `shouldForwardProp` and it was really confusing to me when reading the button component, as I didn’t expect a `<StyledButton>` to do anything but styling ...

Reply:
Yes, 100%. I did this for the button already, but haven't done it here. One issue is that we export styledbutton in some cases and use it as a sub selector in emotion - I need to ensure that that also targets linkbutton now

Reply:
most of the tests here are `findByRole('button').isDisabled()` which no longer works, as links cannot be disabled...